Utils/src/CandidateTrackFilter.cc

```diff
@@ -0,0 +1,380 @@
+		// place some minimal cuts on the tracks
+		//-------------------------------------------------------------------------------------
+		if (!filterOnTrack(pfCand,track)) continue;
+
+		//-------------------------------------------------------------------------------------
+		// fill the track values and PFCandidate values we'd like to save
+		//-------------------------------------------------------------------------------------
+		auto transientTrack = ttBuilder->build(track);
+		infos.fill(primaryVertex,pfCand,track,transientTrack,true);
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	// loop over lostTracks
+	//-------------------------------------------------------------------------------------------------
+	for (size_t i=0; i<lostTracks->size();i++) {
+		const pat::PackedCandidate lostTrack = (*lostTracks)[i];
+
+		//-------------------------------------------------------------------------------------
+		// pull the track details, if available
+		//-------------------------------------------------------------------------------------
+		if (!lostTrack.hasTrackDetails()) continue;
+		auto track = lostTrack.pseudoTrack();
+
+		//-------------------------------------------------------------------------------------
+		// place some minimal cuts on the tracks
+		//-------------------------------------------------------------------------------------
+		if (!filterOnTrack(lostTrack,track)) continue;
+
+		//-------------------------------------------------------------------------------------
+		// fill the track values and PFCandidate values we'd like to save
+		//-------------------------------------------------------------------------------------
+		auto transientTrack = ttBuilder->build(track);
+		infos.fill(primaryVertex,lostTrack,track,transientTrack,false);
+	}
+
+	//-------------------------------------------------------------------------------------
+	// put track/PFCandidate values back into event
+	//-------------------------------------------------------------------------------------
+	bool result = infos.trks->empty();
+	infos.put(iEvent);
+	return result;
+}
+
+// ------------ other methods  ------------
+bool CandidateTrackFilter::filterOnTrack(const pat::PackedCandidate & pfCand, const reco::Track & track) const {
+	//-------------------------------------------------------------------------------------
+	// only store PFCandidate values if pt >= minPt and |eta| <= maxEta_
+	//-------------------------------------------------------------------------------------
+	if ( (track.pt() < minPt_) || (fabs(track.eta()) > maxEta_) ) return false;
+
+	//-------------------------------------------------------------------------------------
+	// cut on impact parameter quantities
+	//-------------------------------------------------------------------------------------
+	if ( (std::abs(track.dz()) > maxdz_) || (std::abs(track.dxy()) > maxdxy_) ) return false;
```

Comment:
this whole function could just be:
```C++
return track.pt() >= minPt_ and std::abs(track.eta()) <= maxEta_ and ...
```

Reply:
Yes, but I wanted to leave room to add other cuts/selections is need be. Putting it all into one line is not by preferred approach. Are you insisting on that format?

Reply:
`if bool return bool` is a redundant pattern. You can write a single return statement on multiple lines if desired.

Reply:
Okay, this has been changed.